src/dmd/iasmdmd.d

```diff
@@ -4452,6 +4503,11 @@ void asm_primary_exp(out OPND o1)
                             break;
                         }
                     }
+                    //if (tryExpressionToConstantOperand(e, o1))
```

Comment:
There was an unexpected side effect in the new function. I'll come back to this later.